src/rust/bitbox-aes/src/lib.rs

```diff
@@ -0,0 +1,3065 @@
+// Copyright 2025 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_std]
+
+#[cfg_attr(test, macro_use)]
+extern crate alloc;
+
+use alloc::vec::Vec;
+
+use hmac::{digest::FixedOutput, Mac, SimpleHmac};
+use sha2::Sha256;
+
+// AES block size.
+const BLOCK_SIZE: usize = 16;
+
+fn pad_length(buf_len: usize) -> usize {
+    BLOCK_SIZE - buf_len % BLOCK_SIZE
+}
+
+fn pkcs7_pad(buf: &mut Vec<u8>) {
+    let pad_len = pad_length(buf.len());
+    buf.resize(buf.len() + pad_len, pad_len as u8);
+}
+
+fn pkcs7_unpad(buf: &mut Vec<u8>) -> Result<(), ()> {
```

Comment:
If you are returning an error with 0 information you might as well panic.

Reply:
An `()` error is absolutely not the same thing as a panic. The caller will know there was an error and can handle it.

Not sure what the point is of adding more info if the caller does not care about which error path it was, can you explain?

Reply:
How will the user code handle the error if it doesn't know what path failed?